seed/seed20.go

```diff
@@ -436,13 +436,18 @@ func (s *seed20) LoadMeta(mode string, tm timings.Measurer) error {
 }
 
 func (s *seed20) LoadEssentialMeta(essentialTypes []snap.Type, tm timings.Measurer) error {
-	filterEssential := essentialSnapTypesToModelFilter(essentialTypes)
+	var filterEssential func(*asserts.ModelSnap) bool
+	if len(essentialTypes) != 0 {
+		filterEssential = essentialSnapTypesToModelFilter(essentialTypes)
+	} else {
+		filterEssential = essentialSnapTypesToModelFilter([]snap.Type{snap.TypeSnapd, snap.TypeKernel, snap.TypeBase})
```

Comment:
isn't this missing gadget?, also wouldn't setting filterEssentil to nil do the same in this case?

Reply:
Yes you right! And then filter can be nil indeed (that actually confused me because forgetting gadget & nil filter led to returning more snaps that I expected in the test). Thanks.